returnn/torch/data/pipeline.py

```diff
@@ -383,6 +403,21 @@ def __iter__(self):
     def __getitem__(self, index):
         raise Exception(f"{self.__class__.__name__}.__getitem__ is not supported")
 
+    def set_seed(self, seed: int) -> BucketOrderingIterDataPipe:
+        """
+        Sets the seed for the next invocation of ``__iter__``, for compatibility with
+        ``torch.utils.data.graph_settings.apply_random_seed``.
+        """
+        self._seed = abs(seed) % (2**32)  # seed must be within [0, 2**32) for seeding RandomState
+        return self
+
+    def reset(self):
+        """resets the internal state of the data pipe"""
+        if self._seed is None:
+            self._seed = abs(int(torch.empty((), dtype=torch.int32).random_().item()))
```

Comment:
Now you are restricting yourself to 0 to 2**31, while you could get 0 to 2**32-1 by doing this:
```suggestion
            self._seed = int(2**31 + torch.empty((), dtype=torch.int32).random_().item()))
```